### PR TITLE
I77 remove rinoh_stylesheet

### PR DESCRIPTION
**Pull request by alexfargus**
Picking off the issues in #77 this PR removes the ability to set the stylesheet from conf.py. Instead it is read from the template configuration.

* Stylesheet (and pygments stylesheet) cannot be set from conf.py
* Template.rtt files added for regression tests to specify required stylesheet
* Documentation updated
* Deprecation warning is issued when `rinoh_stylesheet` is defined.

### CODE REVIEW

File: src/rinoh/frontend/sphinx/__init__.py [hunk 1]
```diff
@@ -343,13 +327,19 @@ def default_domain_indices(config):
     return config.latex_domain_indices
 
 
+def deprecation_warnings(config, logger):
+    if config.rinoh_stylesheet:
+        logger.warning("'rinoh_stylesheet' has been deprecated. Configure"
+                       " the stylesheet in the document template")
```
Comment (brechtm) [inline src/rinoh/frontend/sphinx/__init__.py:333]: ... template configuration

File: src/rinoh/frontend/sphinx/__init__.py [hunk 2]
```diff
@@ -269,23 +270,6 @@ def template_from_config(config, confdir, warn):
...
-    if config.pygments_style is not None:
-        if stylesheet is not None:
-            base = stylesheet
-        elif 'base' in template_cfg:
-            base = template_cfg['base']['stylesheet']
-        else:
-            base = template_cls.stylesheet.default_value
-        stylesheet = pygments_style_to_stylesheet(config.pygments_style, base)
-    if stylesheet is not None:
-        template_cfg['stylesheet'] = stylesheet
```
Comment (brechtm) [inline src/rinoh/frontend/sphinx/__init__.py:288]: So glad this will be gone! 😄

File: src/rinoh/frontend/sphinx/__init__.py (hunk 1 above)
Comment (alexfargus) [reply src/rinoh/frontend/sphinx/__init__.py]: Fixed.

**Comment by alexfargus** [comment]
Although it seems unrelated, I have added unit tests for the Attribute class because the coverage report indicates that my changes caused a code branch within this class not to be covered.

https://codecov.io/gh/brechtm/rinohtype/compare/abfdc323b7d6b2b717ab3e0c1a5f0b256102cf12...f23f3b502ca51e38d5cc6866c3566ad9361ef84b/changes

**Comment by alexfargus** [comment]
@brechtm I believe that all review points have been addressed and the build is green.

Once this PR is merged I will continue work on removing rinoh_papersize.

**Comment by brechtm** [comment]
Thanks Alex, merged! And thank you for going the extra mile in adding the Attribute tests. You didn't have to do that.

The email notifications from your earlier redactions to this PR kind of got lost in my inbox. Don't hesitate to nudge me if you feel I'm taking a while to react. Cheers!

**Pull request merged**